File: src/test/java/com/ggardet/ModularityTests.java
package com.ggardet;

import org.junit.jupiter.api.Test;
import org.springframework.modulith.core.ApplicationModules;
import org.springframework.modulith.docs.Documenter;

class ModularityTests {
    ApplicationModules modules = ApplicationModules.of(SpringModulithApplication.class);

    @Test
    void shouldBeCompliant() {
        modules.verify();
    }

    @Test
    void writeDocumentationSnippets() {
        new Documenter(modules, "doc")
                .writeModuleCanvases()
                .writeModulesAsPlantUml()
                .writeIndividualModulesAsPlantUml();
    }
}

File: src/test/kotlin/com/ggardet/ModularityTests.kt
package com.ggardet

import org.junit.jupiter.api.Test
import org.springframework.modulith.core.ApplicationModules
import org.springframework.modulith.docs.Documenter

internal class ModularityTests {
    private val modules = ApplicationModules.of(SpringModulithApplication::class.java)

    @Test
    fun shouldBeCompliant() {
        modules.verify()
    }

    @Test
    fun writeDocumentationSnippets() {
        Documenter(modules, "doc")
            .writeModuleCanvases()
            .writeModulesAsPlantUml()
            .writeIndividualModulesAsPlantUml()
    }
}
